spec/api.spec.js

```diff
@@ -123,5 +123,63 @@ describe('Editable', function () {
         cursor.triggerChange()
       })
     })
+
+    describe('findClosestCursorOffset:', function () {
+    /*
+Cursor1:                     | (left: 130)
+Comp 1:   Cristiano Ronaldo wurde 2018 mit grossem Tamtam nach Turin geholt.
+Comp 2:   Der Spieler blieb bei fünf Champions-League-Titeln stehen.
+Cursor 2:                    | (offset: 19 chars)
+    */
+      it('finds the index in a text node', function () {
+        $div.html('Der Spieler blieb bei fünf Champions-League-Titeln stehen.')
+        const {wasFound, offset} = editable.findClosestCursorOffset({
+          element: $div[0],
+          origCoordinates: {top: 0, left: 130}
+        })
+        expect(wasFound).toEqual(true)
+        expect(offset).toEqual(19)
+      })
+
+      /*
+Cursor1:                      | (left: 130)
+Comp 1:   Cristiano Ronaldo wurde 2018 mit grossem Tamtam nach Turin geholt.
+Comp 2:   <p>Der <em>Spieler</em> blieb bei fünf <span>Champions-League-Titeln</span> stehen.</p>
+Cursor 2:                                   |
+    */
+      it('finds the index in a nested html tag structure', function () {
+        $div.html('<p>Der <em>Spieler</em> blieb bei fünf <span>Champions-League-Titeln</span> stehen.</p>')
+        const {wasFound, offset} = editable.findClosestCursorOffset({
+          element: $div[0],
+          origCoordinates: {top: 0, left: 130}
+        })
+        expect(wasFound).toEqual(true)
+        expect(offset).toEqual(19)
+      })
+
+      it('returns not found for empty nodes', function () {
+        $div.html('')
+        const {wasFound} = editable.findClosestCursorOffset({
+          element: $div[0],
+          origCoordinates: {top: 0, left: 130}
+        })
+        expect(wasFound).toEqual(false)
+      })
+
+      /*
+Cursor1:                                                   |
+Comp 1:   Cristiano Ronaldo wurde 2018 mit grossem Tamtam nach Turin geholt.
+Comp 2:   Foo
+Cursor 2: not found
+    */
+      it('returns not found for coordinates that are out of the text area', function () {
+        $div.html('Foo')
```

Comment:
Shouldn't it return an offset at the end of the text here? Like `offset: 3`?

Reply:
IMO it's better if it tells me that this is not really a fit at all. But you already mentioned that you would return the closest cursor position plus the pixel offset. That is fine too, then I can maintain the behavior in the framework. Important IMO:
- if you switch from a main area paragraph say in the middle to a sidebar title I don't want it to try and match a position nor to put the cursor at the end, but putting  the cursor at the beginning